src/SFML/Graphics/Text.cpp

```diff
@@ -235,6 +279,44 @@ void Text::setOutlineThickness(float thickness)
 }
 
 
+////////////////////////////////////////////////////////////
+void Text::setUnderlineColor(const Color& color)
+{
+    m_underlineFillColor = color;
+
+    // Change vertex colors directly, no need to update whole geometry
+    // (if geometry is updated anyway, we can skip this step)
+    if (!m_geometryNeedUpdate)
+    {
+        bool isUnderlined    = m_style & Underlined;
+        bool isStrikeThrough = m_style & StrikeThrough;
+        for (std::size_t i = m_vertices.getVertexCount() - 6 * isUnderlined - 6 * isStrikeThrough;
+             i < m_vertices.getVertexCount();
+             ++i)
+            m_vertices[i].color = color;
+    }
+}
```

Comment:
As ugly as this API gets, this should be called `setUnderlineFillColor` for consistency with all the other fill + outline pairs.

Reply:
Thanks for the feedback! I'll do my part.

Reply:
Ok, fixed it.